src/NonBlocking/ConcurrentDictionary/DictionaryImpl`3.cs

```diff
@@ -745,10 +745,10 @@ internal sealed override TValue GetOrAdd(TKey key, Func<TKey, TValue> valueFacto
 
             // prev value is null or dead.
             // let's try install new value
-            newValObj = newValObj ?? ToObjectValue(result = valueFactory(key));
+            newValObj ??= ToObjectValue(result = valueFactory(key));
```

Comment:
Need to make sure C# does not introduce a temp here. 

There is some analysis in compound operators like `+=` to introduce a temp only when the variable can be modified - i.e. when the variable is not a local or we do something like `v += foo(ref v)`. 
I am less familiar with `&&=`. Hopefully it does the same.

Reply:
Yes, if the operand in question is an instance field, the way it is lowered to IL becomes important. But since `newValObj` is a local, it's not an issue here. Codegen produced appears to be [identical](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYmpBkIYBvGgzNNdxFAwCyACgjAAVjDAYA/AwXZJHGAEpTcxNqczCvHz8GAF4I3xgGd08uGAB3O38NUPCzIJyRLlwOATtveMy8hgBfSsqdJmsAOQdnVw84v0DssxCcszKopNiU9Iru8MqwgqKSgYCssJrxyoBtGxgMAAsIABMASXFJO3Wt3YOxSQB5MT4IQpZGiD2uSV4uN4BzfwBdSrEoXjeDAJKwGO4zGAtFxuDoBYzVWrjf6A7DApjkJAMYAQCCSWF7XAANUiMAAKgBPMQwAA8pIAfBkYnSGBhKTAIAAzOyk/wsAnE+IUqlZKpAA=) while [IL appears to be better](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAmARgFgAoQgBgAJDSA6AJQFcA7DbAWxiYDCEHgAc8MKAGUJAN2xgYAZwDcVKoQDMDYnQF0A3lTrGGWwijoBZABQRgAKxhgMAfjoyAhrjYwAlEZNDShMQ9y8fOgBeMO8YOhc3DhgAd2tfVWDQ4wCsoQ5FNj5rT1j0nLoAX3LyzQYLADlbBydXGJ9/TOMgrOMSiITopNSyztDykLyCor6/DJCq0fKAbUsYDAALCAATAElRXGtVje29kVwAeRFuCHymeogdjlxsDheAc18AXXKRKGxPDBxcy6G5TGBNRzONp+AyVaqjX7/DyAhikJB0YAQCC4aE7RQANXCMAAKgBPEQwAA8xIAfGkojS6BhyTAIAAzazE3xMPGE2JkikZCpAA) for compound assignment (not like it matters).